packages/fxa-shared/metrics/amplitude-event.1.schema.json

```diff
@@ -180,16 +184,25 @@
     "user_properties"
   ],
   "anyOf": [{ "required": ["user_id"] }, { "required": ["device_id"] }],
-  "if": {
-    "$comment": "For event_type's matching the pattern below, make language a required field",
-    "properties": {
-      "event_type": {
-        "type": "string",
-        "pattern": "^fxa_pay_setup -|fxa_pay_subscription_change -"
+  "allOf": [
+    {
+      "if": {
+        "$comment": "For event_type's matching the pattern below, make language a required field",
+        "properties": {
+          "event_type": {
+            "type": "string",
+            "pattern": "^fxa_pay_setup -|fxa_pay_subscription_change -"
+          }
+        }
+      },
+      "then": {
+        "required": ["language"],
+        "properties": {
+          "country_iso": {
+            "pattern": "^[A-Z]{2}$"
+          }
+        }
```

Comment:
I'm sorry to do this, but upon further reflection from our conversation this morning, I think we should validate this property at the top level. It's optional, and a brand new property, so no one will run into issues unless and until they add this property to their events. And if they do, we want it to be consistent everywhere.

If we have the pattern buried down here, it makes it look like `country_iso` is just a string at first glance.

I've got the diff for you to apply and already tested; it should require very minimal changes to your existing tests. Just need to update the validation error message to remove the `then` part.

```diff
diff --git a/packages/fxa-shared/metrics/amplitude-event.1.schema.json b/packages/fxa-shared/metrics/amplitude-event.1.schema.json
index 58ec9d120a..6cec306589 100644
--- a/packages/fxa-shared/metrics/amplitude-event.1.schema.json
+++ b/packages/fxa-shared/metrics/amplitude-event.1.schema.json
@@ -58,7 +58,8 @@
     },
     "country_iso": {
       "description": "The 2 character country ISO code",
-      "type": "string"
+      "type": "string",
+      "pattern": "^[A-Z]{2}$"
     },
     "country": {
       "description": "The localized full country name where the locale is based (indirectly) on a request's `Accept-Language` header.",
@@ -196,12 +197,7 @@
         }
       },
       "then": {
-        "required": ["language"],
-        "properties": {
-          "country_iso": {
-            "pattern": "^[A-Z]{2}$"
-          }
-        }
+        "required": ["language"]
       }
     }
   ]
```